hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestXceiverServerDomainSocket.java

```diff
@@ -0,0 +1,713 @@
+      dataBuf.flip();
+      readSize = dataIn.read(dataBuf);
+      assertEquals(-1, readSize);
+
+      // check metrics
+      assertEquals(1, containerMetrics.getContainerLocalOpsMetrics(ContainerProtos.Type.GetBlock));
+    } finally {
+      factory.close();
+      if (sock != null) {
+        try {
+          sock.close();
+        } catch (IOException e) {
+        }
+      }
+      server.stop();
+    }
+  }
+
+  /**
+   * Test server is not listening.
+   */
+  @Test
+  @Timeout(30)
+  public void testServerNotListening() {
+    conf.set(OzoneClientConfig.OZONE_DOMAIN_SOCKET_PATH, new File(dir, "ozone-socket").getAbsolutePath());
+    DomainSocketFactory factory = DomainSocketFactory.getInstance(conf);
+    DomainSocket sock = null;
+    try {
+      sock = factory.createSocket(readTimeout, writeTimeout, localhost);
+    } catch (IOException e) {
+      assertTrue(e instanceof ConnectException);
+      assertTrue(e.getMessage().contains("connect(2) error: No such file or directory"));
+    } finally {
+      factory.close();
+      if (sock != null) {
+        try {
+          sock.close();
+        } catch (IOException e) {
+        }
+      }
+    }
+  }
+
+  /**
+   * Test server is not started to accept new connection.
+   * Although socket can be created, read will fail, write can succeed.
+   */
+  @Test
+  @Timeout(30)
+  public void testServerNotStart() {
+    conf.set(OzoneClientConfig.OZONE_DOMAIN_SOCKET_PATH, new File(dir, "ozone-socket").getAbsolutePath());
+    DomainSocketFactory factory = DomainSocketFactory.getInstance(conf);
+    XceiverServerDomainSocket server = new XceiverServerDomainSocket(MockDatanodeDetails.randomDatanodeDetails(),
+        conf, null, readExecutors, metrics, factory);
+    DomainSocket sock = null;
+    try {
+      sock = factory.createSocket(readTimeout, writeTimeout, localhost);
+      assertTrue(sock.isOpen());
+      // send request
+      final DataOutputStream  outputStream = new DataOutputStream(sock.getOutputStream());
+      outputStream.writeShort(OzoneClientConfig.DATA_TRANSFER_VERSION);
+      outputStream.writeShort(GetBlock.getNumber());
+      getBlockRequest().writeDelimitedTo(outputStream);
+      outputStream.flush();
+
+      final DataInputStream inputStream = new DataInputStream(sock.getInputStream());
+      inputStream.readShort();
+    } catch (IOException e) {
+      assertTrue(e instanceof SocketTimeoutException);
+      assertTrue(e.getMessage().contains("read(2) error: Resource temporarily unavailable"));
+    } finally {
+      factory.close();
+      if (sock != null) {
+        try {
+          sock.close();
+        } catch (IOException e) {
+        }
+      }
+      server.stop();
+    }
+  }
+
+  @Test
+  @Timeout(30)
+  public void testReadTimeout() throws InterruptedException {
+    conf.set(OzoneClientConfig.OZONE_DOMAIN_SOCKET_PATH, new File(dir, "ozone-socket").getAbsolutePath());
+    conf.set(OzoneConfigKeys.OZONE_CLIENT_READ_TIMEOUT, "5s");
```

Comment:
Can we decrease this to 2s to reduce `sleep()` time without introducing flakiness?